docs/dev/perl5/index.html

```diff
@@ -27,7 +27,7 @@ <h3>Releases</h3>
     <ul>
       <li>
         Latest development release
-        is <a href="https://metacpan.org/pod/release/CORION/perl-5.37.6/pod/perl.pod">Perl 5.37.6</a>
+        is <a href="https://metacpan.org/release/HYDAHY/perl-5.37.7/view/pod/perl.pod">Perl 5.37.7</a>
```

Comment:
I realize this has been this way since changing the link to metacpan.org, but I don't think this is actually a great link.  It links to the documentation, which has nothing particularly 5.37.7 specific.  Should it link to the download link instead?  Is there another page that would be more useful?

Reply:
Yes, IMO it should link to https://metacpan.org/release/HYDAHY/perl-5.37.7 instead.

Reply:
I'm leaning towards https://metacpan.org/release/HYDAHY/perl-5.37.7/view/pod/perldelta.pod

Reply:
I can definitely see the perldelta for the release being of most interest to the types of users who are interested in development releases.

I've updated the PR to point to the one at _perldoc.perl.org_, based on Grinnz's argument on #p5p that it has better rendering of linked pages, but I'd be happy to change to https://metacpan.org/release/HYDAHY/perl-5.37.7/view/pod/perldelta.pod if preferred.